sympy/geometry/line.py

```diff
@@ -2353,8 +2354,8 @@ def __new__(cls, p1, pt=None, direction_ratio=[], **kwargs):
 
         return LinearEntity3D.__new__(cls, p1, pt, **kwargs)
 
-    def equation(self, x='x', y='y', z='z', k='k'):
```

Comment:
This should get a deprecation. I think the way to handle it would be to make k=None. Then, if k is not None in the routine, the warning would be raised indicating that the routine no longer needs k and that that argument will be removed in the future.

Reply:
@smichr I didn't understand what you are saying. Should I create a warning if k is not None ?

Reply:
Search the repo for the word 'deprecation' and you will se examples.

Reply:
@smichr are you talking about https://github.com/briancurtin/deprecation , http://deprecation.readthedocs.io/en/latest/ ? And are talking about using @deprecated and @fail_if_not_removed ? if yes then can you guide me how to implement it ?

Reply:
At sympy/core/numbers.py line 943 there is a test for a no-longer supported keyword `prec`. Can you use that as an example for `k`?

Reply:
You will need to use `deprecated_since=1.2` argument as well.